src/multisketch.rs

```diff
@@ -196,6 +191,67 @@ impl MultiSketch {
 
         self
     }
+    pub fn remove_metadata(
+        &mut self,
+        output_file_name: &str,
+        genome_ids_to_remove: &[String],
+    ) -> std::io::Result<()> {
+        println!("{}", self);
+        let mut new_sketch_metadata: Vec<Sketch> = Vec::with_capacity(self.sketch_metadata.len());
+
+        for sketch in &self.sketch_metadata {
+            if !genome_ids_to_remove.contains(&(*sketch.name()).to_string()) {
+                new_sketch_metadata.push(sketch.clone());
+            }
+        }
+        self.sketch_metadata = new_sketch_metadata;
+        let _ = self.save_metadata(output_file_name);
+        Ok(())
+    }
+
+    pub fn remove_genomes(
+        &mut self,
+        input_prefix: &str,
+        output_file: &str,
+        genome_ids_to_remove: &[String],
+    ) -> std::io::Result<()> {
```

Comment:
You'd be better returning an Error type in this function, easiest to use `anyhow::Error`. Then you return `Ok(())` if it all works, but return an Error if `write_batch()` errors, or if there were missing IDs